include/dali/core/tensor_shape.h

```diff
@@ -812,9 +811,8 @@ struct TensorListShape<DynamicDimensions>
   template <int other_sample_ndim>
   TensorListShape &operator=(TensorListShape<other_sample_ndim> &&other) {
     shapes = std::move(other.shapes);
-    nsamples = other.size();
-    ndim = other.sample_dim();
-    other.nsamples = 0;
+    nsamples = std::exchange(other.nsamples, 0);
+    ndim = other.sample_dim();  // Sample dim could be static, and it is valid to not zero it.
```

Comment:
IIUC, previously the nsamples wasn't moved, but passed by value, so this changes the behaviour. Is that intentional?

Reply:
As above.